### PR TITLE
feat(expressions): Adding the ability to specify a runtime spel modifier class

### PR DESCRIPTION
**Pull request by jeyrschabu**
- added runtime expression updaters
- parameters.foo -> trigger.parameters.foo after parameters have been processed once

### CODE REVIEW

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy [hunk 1]
```diff
@@ -449,6 +458,25 @@ abstract class ContextUtilities {
 
 }
 
+/**
+ * Include the execution object (#root.execution) for Stage locating functions at runtime
+ * @param expression #stage('property') becomes #stage(#root.execution, 'property')
+ * @return an execution aware helper function (only limited to judg(e?)ment & stage)
+ */
+class ExecutionAwareExpressionUpdater implements RuntimeExpressionUpdater {
```
Comment (jeyrschabu) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy]: Externalized this fit into the pattern.  Was previously a standalone method

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt [hunk 2]
```diff
@@ -62,6 +63,21 @@ interface ExpressionAware {
     return this
   }
 
+  /**
```
Comment (jeyrschabu) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt]: The idea here is that once `parameters` have been processed, they are inserted into `trigger`.
This helps route the call to the right place using the `RuntimeExpressionUpdater` pattern.

**Comment by jeyrschabu** [comment]
The second part of this PR is to route parameter accessing expressions to the right place once they have been processed. Once parameters have been processed, they are inserted in `trigger`.
This helps route the call to the right place using the RuntimeExpressionUpdater pattern.

**Comment by jeyrschabu** [comment]
@spinnaker/netflix-reviewers PTAL

**Comment by robzienert** [comment]
@jeyrschabu Should we close?

**Comment by jeyrschabu** [comment]
Yeah. closing.

**Pull request closed without merging**